### PR TITLE
Created a calculator and wrote tests for it

### PR DESCRIPTION
**Pull request by EvilCrampus**

### CODE REVIEW

File: src/main/java/Calculate.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+                return getTheSubtractionResult(firstNumber, secondNumber);
+            case DIVISION:
+                return getTheDivisionResult(firstNumber, secondNumber);
+            case MULTIPLICATION:
+                return getTheMultiplicationResult(firstNumber, secondNumber);
+            case EXPONENTIATION:
+                return getTheExponentiationResult(firstNumber, secondNumber);
+            default:
+                throw new NoSuchElementException("Incorrect operation");
+
```
Comment (Yehor-Zapara) [inline src/main/java/Calculate.java:24]: We don't need whitespace here

File: src/test/java/CalculateTest.java [hunk 2]
```diff
@@ -0,0 +1,226 @@
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import java.util.NoSuchElementException;
+import org.junit.jupiter.api.Test;
+
+class CalculateTest {
+    private static final Calculate calculate = new Calculate();
```
Comment (Yehor-Zapara) [inline src/test/java/CalculateTest.java:8]: Let's rename this constant, because when we write **calculate.calculate** it's confusing. And also constants should be with only uppercase chars and _.

File: src/test/java/CalculateTest.java [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+
+import java.util.NoSuchElementException;
+import org.junit.jupiter.api.Test;
+
+class CalculateTest {
+    private static final Calculate calculate = new Calculate();
+
+    @Test
+    void additionWithTwoPositiveOperands_Ok() {
+        double actual = calculate.calculate(6, 8, Calculate.PLUS);
```
Comment (Yehor-Zapara) [inline src/test/java/CalculateTest.java:12]: Let's create fields **answer** and **actual** in class and in tests we only initialize them

File: src/test/java/CalculateTest.java [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+        double answer = 2;
+        assertEquals(answer, actual);
+    }
+
+    @Test
+    void additionWithZeroInDifferentPlaces_Ok() {
+        double actual = calculate.calculate(0, 8, Calculate.PLUS);
+        double answer = 8;
+        assertEquals(answer,actual);
+
```
Comment (Yehor-Zapara) [inline src/test/java/CalculateTest.java:36]: I don't think that it is good to use whitespaces between tests in one method. Maybe is not a mistake, and that is more readable, better to ask a mentor.

**Review by Yehor-Zapara — CHANGES_REQUESTED** [review]

File: src/main/java/Calculate.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+            case MINUS:
+                return getTheSubtractionResult(firstNumber, secondNumber);
+            case DIVISION:
+                return getTheDivisionResult(firstNumber, secondNumber);
+            case MULTIPLICATION:
+                return getTheMultiplicationResult(firstNumber, secondNumber);
+            case EXPONENTIATION:
+                return getTheExponentiationResult(firstNumber, secondNumber);
+            default:
+                throw new NoSuchElementException("Incorrect operation");
```
Comment (MaliukDaria) [inline src/main/java/Calculate.java:23]: Maybe IllegalArgumentException would be better here?
image

File: src/test/java/CalculateTest.java [hunk 6]
```diff
@@ -0,0 +1,228 @@
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import java.util.NoSuchElementException;
+import org.junit.jupiter.api.Test;
+
+class CalculateTest {
+    private static final Calculate CALCULATOR = new Calculate();
```
Comment (MaliukDaria) [inline src/test/java/CalculateTest.java:8]: I dont think we should make calculator as constant. Lets make it as private static variable and initialize it in @BeforeAll method

File: src/test/java/CalculateTest.java [hunk 7]
```diff
@@ -0,0 +1,228 @@
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import java.util.NoSuchElementException;
+import org.junit.jupiter.api.Test;
+
+class CalculateTest {
+    private static final Calculate CALCULATOR = new Calculate();
+    private double actual;
+    private double answer;
```
Comment (MaliukDaria) [inline src/test/java/CalculateTest.java:10]:
```suggestion
    private double expected;

```

`answer` less informative name

File: src/test/java/CalculateTest.java [hunk 8]
```diff
@@ -0,0 +1,228 @@
...
+import java.util.NoSuchElementException;
+import org.junit.jupiter.api.Test;
+
+class CalculateTest {
+    private static final Calculate CALCULATOR = new Calculate();
+    private double actual;
+    private double answer;
+
+    @Test
+    void additionWithTwoPositiveOperands_Ok() {
```
Comment (MaliukDaria) [inline src/test/java/CalculateTest.java:13]: Let's name our methods based on the pattern `<methodUnderTest>_<state>_<expectedBehavior>`

File: src/test/java/CalculateTest.java (hunk 4 above)
Comment (MaliukDaria) [reply src/test/java/CalculateTest.java:38]: Agree, lets remove redundant empty lines

File: src/test/java/CalculateTest.java [hunk 9]
```diff
@@ -0,0 +1,228 @@
...
+        actual = CALCULATOR.calculate(-6, -8, Calculate.MINUS);
+        answer = 2;
+        assertEquals(answer, actual);
+    }
+
+    @Test
+    void subtractionWithPositiveAndNegativeOperands_Ok() {
+        actual = CALCULATOR.calculate(-6, 8, Calculate.MINUS);
+        answer = -14;
+        assertEquals(answer, actual);
```
Comment (MaliukDaria) [inline src/test/java/CalculateTest.java:66]: we can add to this test check for first negative and second positive operands

File: src/test/java/CalculateTest.java [hunk 10]
```diff
@@ -0,0 +1,228 @@
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import java.util.NoSuchElementException;
+import org.junit.jupiter.api.Test;
+
+class CalculateTest {
```
Comment (MaliukDaria) [inline src/test/java/CalculateTest.java:7]: what about `addition for min and max double values;`?

**Review by MaliukDaria — CHANGES_REQUESTED** [review]

**Review by MaliukDaria — CHANGES_REQUESTED** [review]
Travis failed....

**Review by MaliukDaria — APPROVED** [review]
